src/core/providers/ogr/qgscplerrorhandler_p.h

```diff
@@ -45,7 +50,64 @@ class QgsCPLErrorHandler
     QgsCPLErrorHandler( const QgsCPLErrorHandler &other ) = delete;
     QgsCPLErrorHandler &operator=( const QgsCPLErrorHandler &other ) = delete;
 
+  private:
+
+    QString mSourceName;
+
+};
+
+/**
+ * \ingroup core
+ * \class QgsCPLErrorCollectorHandler
+ *
+ * A GDAL error handler which collects errors for later processing.
+ *
+ * \since QGIS 3.28
+ */
+class QgsCPLErrorCollectorHandler
+{
+    static void CPL_STDCALL showError( CPLErr errClass, int, const char *msg )
+    {
+      if ( errClass != CE_None )
+      {
+        QStringList *errors = reinterpret_cast< QStringList * >( CPLGetErrorHandlerUserData() );
+        if ( errors )
+        {
+          errors->append( QString( msg ) );
+        }
+      }
+    }
+
+  public:
+    QgsCPLErrorCollectorHandler()
+    {
+      CPLPushErrorHandlerEx( showError, reinterpret_cast< void * >( &mErrors ) );
```

Comment:
doesn't the following just work ?
```suggestion
      CPLPushErrorHandlerEx( showError, &mErrors );
```